modules/tensor_mechanics/src/actions/LineElementAction.C

```diff
@@ -0,0 +1,827 @@
+      }
+    }
+
+    params = _factory.getValidParams("NewmarkVelAux");
+    params.applyParameters(parameters(), {"boundary"});
+    params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+
+    for (unsigned i = 0; i < 2 * _ndisp; ++i)
+    {
+      std::string aux_kernel_name = name() + "_newmark_vel_" + Moose::stringify(i);
+
+      if (i < _ndisp)
+      {
+        params.set<AuxVariableName>("variable") = _velocities[i];
+        params.set<std::vector<VariableName>>("acceleration") = {_accelerations[i]};
+        _problem->addAuxKernel("NewmarkVelAux", aux_kernel_name, params);
+      }
+      else
+      {
+        params.set<AuxVariableName>("variable") = _rot_velocities[i - _ndisp];
+        params.set<std::vector<VariableName>>("acceleration") = {_rot_accelerations[i - _ndisp]};
+        _problem->addAuxKernel("NewmarkVelAux", aux_kernel_name, params);
+      }
+    }
+  }
+}
+
+void
+LineElementAction::actAddNodalKernels()
+{
+  if (_current_task == "add_nodal_kernel" && !_truss)
+  {
+    // NodalTranslationalInertia and NodalRotattionalInertia currently accept only constant real
+    // numbers for eta
+    Real eta = 0.0;
+    if (_dynamic_nodal_rotational_inertia || _dynamic_nodal_translational_inertia)
+    {
+      std::istringstream ss(getParam<MaterialPropertyName>("eta"));
+      Real real_value;
+
+      if (ss >> real_value && ss.eof())
+        eta = real_value;
+      else
+        mooseError("LineElementAction: When using nodal mass/inertia matrices, `eta` can currently "
+                   "only take constant real values");
+    }
+
+    if (_dynamic_nodal_translational_inertia)
+    {
+      auto params = _factory.getValidParams("NodalTranslationalInertia");
+      params.applyParameters(parameters(),
+                             {"save_in", "diag_save_in", "use_displaced_mesh", "eta"});
+      params.set<Real>("mass") = getParam<Real>("nodal_mass");
+      params.set<Real>("eta") = eta;
+      params.set<bool>("use_displaced_mesh") = _use_displaced_mesh;
+
+      for (unsigned i = 0; i < _ndisp; ++i)
+      {
+        std::string nodal_kernel_name =
+            name() + "_nodal_translational_inertia_" + Moose::stringify(i);
+
+        params.set<NonlinearVariableName>("variable") = _displacements[i];
+        params.set<std::vector<VariableName>>("velocity") = {_velocities[i]};
+        params.set<std::vector<VariableName>>("acceleration") = {_accelerations[i]};
+
+        if (_save_in.size() == 2 * _ndisp)
+          params.set<std::vector<AuxVariableName>>("save_in") = {_save_in[i]};
+
+        if (_diag_save_in.size() == 2 * _ndisp)
+          params.set<std::vector<AuxVariableName>>("diag_save_in") = {_diag_save_in[i]};
+
+        _problem->addNodalKernel("NodalTranslationalInertia", nodal_kernel_name, params);
+      }
+    }
+
+    if (_dynamic_nodal_rotational_inertia)
+    {
+      auto params = _factory.getValidParams("NodalRotationalInertia");
+      params.applyParameters(parameters(),
```

Comment:
I'm looking at these large blocks of conditional logic to apply parameters and I don't think any of it is necessary unless I'm missing something. The default behavior of applyParameters is to only apply parameters with matching names and types to avoid logic like this. Also, it only applies them when the source is valid, etc. 

Unless you are transforming the parameters (new names or augmentation) you should just apply call apply parameters without any filters and just let it do it's thing.

Reply:
UPDATE: OK - Now I see, you are transforming the names. You can ignore this...